block/internal/cache/manager.go

```diff
@@ -0,0 +1,311 @@
+	for _, data := range dataList {
+		if len(data.Txs) == 0 {
+			continue // Skip empty data
+		}
+		// Note: Actual signing needs to be done by the executing component
+		// as it has access to the signer. This method returns unsigned data
+		// that will be signed by the executing component when needed.
+		signedDataList = append(signedDataList, &types.SignedData{
+			Data: *data,
+			// Signature and Signer will be set by executing component
+		})
+	}
+
+	return signedDataList, nil
+}
+
+func (m *implementation) SetLastSubmittedHeaderHeight(ctx context.Context, height uint64) {
+	m.pendingHeaders.setLastSubmittedHeaderHeight(ctx, height)
+}
+
+func (m *implementation) SetLastSubmittedDataHeight(ctx context.Context, height uint64) {
+	m.pendingData.setLastSubmittedDataHeight(ctx, height)
+}
+
+func (m *implementation) GetLastSubmittedHeaderHeight() uint64 {
+	return m.pendingHeaders.getLastSubmittedHeaderHeight()
+}
+
+func (m *implementation) GetLastSubmittedDataHeight() uint64 {
+	return m.pendingData.getLastSubmittedDataHeight()
+}
+
+func (m *implementation) NumPendingHeaders() uint64 {
+	return m.pendingHeaders.numPendingHeaders()
+}
+
+func (m *implementation) NumPendingData() uint64 {
+	return m.pendingData.numPendingData()
+}
+
+// Pending events operations
+func (m *implementation) SetPendingEvent(height uint64, event *DAHeightEvent) {
+	m.pendingEventsCache.SetItem(height, event)
+}
+
+func (m *implementation) GetPendingEvents() map[uint64]*DAHeightEvent {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+
+	events := make(map[uint64]*DAHeightEvent)
+	m.pendingEventsCache.RangeByHeight(func(height uint64, event *DAHeightEvent) bool {
+		events[height] = event
+		return true
+	})
+	return events
+}
+
+func (m *implementation) DeletePendingEvent(height uint64) {
+	m.pendingEventsCache.DeleteItem(height)
+}
+
+func (m *implementation) RangePendingEvents(fn func(uint64, *DAHeightEvent) bool) {
+	m.pendingEventsCache.RangeByHeight(fn)
+}
+
+// Cleanup operations
+func (m *implementation) ClearProcessedHeader(height uint64) {
+	m.headerCache.DeleteItem(height)
+}
+
+func (m *implementation) ClearProcessedData(height uint64) {
+	m.dataCache.DeleteItem(height)
+}
+
+func (m *implementation) SaveToDisk() error {
+	cfgDir := filepath.Join(m.config.RootDir, "data")
+
+	if err := m.headerCache.SaveToDisk(filepath.Join(cfgDir, headerCacheDir)); err != nil {
+		return fmt.Errorf("failed to save header cache to disk: %w", err)
+	}
+
+	if err := m.dataCache.SaveToDisk(filepath.Join(cfgDir, dataCacheDir)); err != nil {
+		return fmt.Errorf("failed to save data cache to disk: %w", err)
+	}
+
+	if err := m.pendingEventsCache.SaveToDisk(filepath.Join(cfgDir, pendingEventsCacheDir)); err != nil {
+		return fmt.Errorf("failed to save pending events cache to disk: %w", err)
+	}
+
+	return nil
+}
+
+func (m *implementation) LoadFromDisk() error {
+	// Register types for gob encoding
+	gob.Register(&types.SignedHeader{})
```

Comment:
why use gob over json, with json we could use the fast version of https://github.com/bytedance/sonic

Reply:
Because gob is smaller and faster.
It is simply go struct written to disk (non human readable however)